test/extended/gssapi.sh

```diff
@@ -0,0 +1,301 @@
+        "headers": ["Remote-User"]
+      }
+    }]
+  }
+}
+'
+cp ${SERVER_CONFIG_DIR}/master/master-config.yaml ${SERVER_CONFIG_DIR}/master/master-config.tmp.yaml
+openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.tmp.yaml --patch="${patch}" > ${SERVER_CONFIG_DIR}/master/master-config.yaml
+export USE_LATEST_IMAGES=true
+start_os_server
+
+export KUBECONFIG="${ADMIN_KUBECONFIG}"
+
+install_registry
+wait_for_registry
+
+oc login -u system:admin
+oc new-project gssapiproxy
+oadm policy add-scc-to-user anyuid -z default -n gssapiproxy
+
+# create all the resources we need
+oc create -f test/extended/testdata/gssapi/proxy
+
+oc create -f test/extended/testdata/gssapi/fedora/base
+oc create -f test/extended/testdata/gssapi/fedora/kerberos
+oc create -f test/extended/testdata/gssapi/fedora/kerberos_configured
+
+# kick off a build and wait for it to finish
+oc start-build --from-file=test/extended/testdata/gssapi/proxy --follow gssapiproxy
+
+# oc set env dc/gssapiproxy-server host='gssapiproxy-server.gssapi.svc.cluster.local' realm='MYDOMAIN.COM' BACKEND='http://openshift.gssapi.svc.cluster.local'
+
+oc set env dc/gssapiproxy-server AUTH=GSSAPI_AND_BASIC
+
+oc deploy gssapiproxy-server
+
+# server_ready_template=(                                  \
+# "{{with \$items := .items}}"                             \
+#     "{{with \$item := index \$items 0}}"                 \
+#         "{{range \$map := \$item.status.conditions}}"    \
+#             "{{with \$state := index \$map \"type\"}}"   \
+#                 "{{\$state}}"                            \
+#             "{{end}}"                                    \
+#             "{{with \$valid := index \$map \"status\"}}" \
+#                 "{{\$valid}} "                           \
+#             "{{end}}"                                    \
+#         "{{end}}"                                        \
+#     "{{end}}"                                            \
+# "{{end}}"                                                \
+# )
+# server_ready_template=$(IFS=$""; echo "${server_ready_template[*]}") # re-formats template for use
+
+# wait for auth proxy to be ready
+os::cmd::try_until_text "oc get pods -l deploymentconfig=gssapiproxy-server -o jsonpath='{.items[*].status.conditions[?(@.type==\"Ready\")].status}'" '^True\n$'
+
+# KERBEROS_SERVICE_IP=$(oc get --output-version=v1beta3 --template="{{ .spec.portalIP }}" service gssapiproxy-server)
+# echo $KERBEROS_SERVICE_IP
+
+# exit 0
+
+# root_patch="{\"spec\":{\"volumes\":[{\"name\":\"os\",\"hostPath\":{\"path\":\"${OS_ROOT}\"}}]}}"
+
+REGISTRY_IP=$(oc get svc docker-registry -n default -o jsonpath='{.spec.clusterIP}:{.spec.ports[0].targetPort}')
+pushd test/extended/testdata/gssapi/fedora
+    oc start-build --from-dir=base --follow fedora-gssapi-base
+    sed "s:OS_ROOT:${OS_ROOT}:" base/fedora-base-pod | \
+    sed "s#REGISTRY_IP#${REGISTRY_IP}#" > ${SERVER_CONFIG_DIR}/fedora-base-pod.json
```

Comment:
if you need to swap values into files containing API definitions, a template with params would be a better choice